sagemaker-python-sdk/mxnet_gluon_mnist/mnist.py

```diff
@@ -181,3 +181,29 @@ def transform_fn(net, data, input_content_type, output_content_type):
     prediction = mx.nd.argmax(output, axis=1)
     response_body = json.dumps(prediction.asnumpy().tolist()[0])
     return response_body, output_content_type
+
+
+if __name__ == '__main__':
+    # mxnet-1.3 requirements on https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/mxnet/README.rst
+    parser = argparse.ArgumentParser()
+
+    # hyperparameters
+    parser.add_argument('--sm-hps', type=json.loads, default=os.environ['SM_HPS'])
+
+    # input data and model directories
+    parser.add_argument('--model-dir', type=str, default=os.environ['SM_MODEL_DIR'])
+    parser.add_argument('--training', type=str, default=os.environ['SM_CHANNEL_TRAINING'])
+
+    # other stuff
+    parser.add_argument('--sm-current-host', type=str, default=os.environ['SM_CURRENT_HOST'])
+    parser.add_argument('--sm-hosts', type=json.loads, default=os.environ['SM_HOSTS'])
+    parser.add_argument('--sm-num-gpus', type=int, default=os.environ['SM_NUM_GPUS'])
```

Comment:
we generally don't keep the "sm" prefix when naming the arguments in argparse, so make these `--current-host`, `--hosts`, and `--num-gpus`. Also `--hyperparameters` above instead of `--sm-hps`